src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java

```diff
@@ -0,0 +1,145 @@
+    private final Deployment deployment;
+    private final Instance instance;
+    private final OtpServer otpServer;
+    private final AmazonEC2 ec2 = AmazonEC2Client.builder().build();
+    private final CloseableHttpClient httpClient = HttpClients.createDefault();
+    // If the job takes longer than XX seconds, fail the job.
+    private static final int TIMEOUT_MILLIS = 60 * 60 * 1000; // One hour
+
+    public MonitorServerStatusJob(String owner, Deployment deployment, Instance instance, OtpServer otpServer) {
+        super(
+            owner,
+            String.format("Monitor server setup %s", instance.getPublicIpAddress()),
+            JobType.MONITOR_SERVER_STATUS
+        );
+        this.deployment = deployment;
+        this.instance = instance;
+        this.otpServer = otpServer;
+        status.message = "Checking server status...";
+    }
+
+    @Override
+    public void jobLogic() {
+        long startTime = System.currentTimeMillis();
+        // Get OTP URL for instance to check for availability.
+        String otpUrl = String.format("http://%s/otp", instance.getPublicIpAddress());
+        boolean otpIsRunning = false, routerIsAvailable = false;
+        // Progressively check status of OTP server
+        if (deployment.buildGraphOnly) {
+            // FIXME No need to check that OTP is running. Just check to see that the graph is built.
+//            FeedStore.s3Client.doesObjectExist(otpServer.s3Bucket, );
+        }
+        // First, check that OTP has started up.
+        status.update("Instance status is OK. Waiting for OTP...", 20);
+        while (!otpIsRunning) {
+            LOG.info("Checking that OTP is running on server at {}.", otpUrl);
+            // If the request is successful, the OTP instance has started.
+            otpIsRunning = checkForSuccessfulRequest(otpUrl, 5);
+            if (System.currentTimeMillis() - startTime > TIMEOUT_MILLIS) {
+                status.fail(String.format("Job timed out while monitoring setup for server %s", instance.getInstanceId()));
+                return;
+            }
+        }
+        String otpMessage = String.format("OTP is running on server %s (%s). Building graph...", instance.getInstanceId(), otpUrl);
+        LOG.info(otpMessage);
+        status.update(otpMessage, 30);
+        // Once this is confirmed, check for the existence of the router, which will indicate that the graph build is
+        // complete.
+        String routerUrl = String.format("%s/routers/default", otpUrl);
+        while (!routerIsAvailable) {
+            LOG.info("Checking that router is available (i.e., graph build or read is finished) at {}", routerUrl);
+            // If the request was successful, the graph build is complete!
+            // TODO: Substitute in specific router ID? Or just default to... default.
+            routerIsAvailable = checkForSuccessfulRequest(routerUrl, 5);
```

Comment:
No timeout in this while loop means this could result in an infinite loop of checking if OTP started up even when it has already failed.